From 8fd9fffb8097132f70b5c83e32a8452e0bfa05c0 Mon Sep 17 00:00:00 2001
From: iAboothahir <aboothahirpkd@gmail.com>
Date: Sat, 8 Oct 2022 16:43:42 +0530
Subject: [PATCH] arm64: dts: qcom: pm660: Add battery fuel gauge

PM660 has a fuel guage gen 3, which is compatible with the one
used in MSM8998.

Signed-off-by: Aboothahir U <aboothahirpkd@gmail.com>

Source-Commit: https://github.com/sdm660-mainline/linux/commit/8fd9fffb8097132f70b5c83e32a8452e0bfa05c0
Scope: SDM660-common
Porting notes: Original PM660 node hunk applies unchanged to Linux 7.3-rc5.
---
diff --git a/arch/arm64/boot/dts/qcom/pm660.dtsi b/arch/arm64/boot/dts/qcom/pm660.dtsi
index 6adbd23fcb0d6b..11b94d883d2c7b 100644
--- a/arch/arm64/boot/dts/qcom/pm660.dtsi
+++ b/arch/arm64/boot/dts/qcom/pm660.dtsi
@@ -198,6 +198,16 @@
 			};
 		};
 
+		pm660_fg: battery@4000 {
+			compatible = "qcom,pmi8998-fg";
+			reg = <0x4000>;
+
+			interrupts = <0x0 0x40 0x3 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "soc-delta";
+
+			status = "disabled";
+		};
+
 		pm660_rradc: adc@4500 {
 			compatible = "qcom,pm660-rradc";
 			reg = <0x4500>;

